File: week 6/sentimental-readability/readability.py
text = input("Text: ")

sentence_count = 0
# counts the number of sentences, looking for how many times a dot, exclamation mark or a question mark has appeared
sentence_count = text.count(".") + text.count("!") + text.count("?")
if sentence_count == 0:  # if no ".", "!" or "?" at the end of sentence, then set the counter to 1
    sentence_count = 1

# creates a list called words and inserts each word of the text as different elements
words = text.split()
word_count = len(words)

# count letters, iterating through each symbol, looking if it is one of the letters of the alphabet
letter_count = 0
for symbol in text:
    if symbol.upper() in "QWERTYUIOPASDFGHJKLZXCVBNM":
        letter_count += 1

# the Coleman-Liau index
l = (letter_count / word_count) * 100
s = (sentence_count / word_count) * 100

grade = round(0.0588 * l - 0.296 * s - 15.8)

# print resultssubmit50 cs50/problems/2024/x/sentimental/readability
if grade < 1:
    print("Before Grade 1")
elif grade > 16:
    print("Grade 16+")
else:
    print(f"Grade {grade}")

File: week 2/readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

void check_grade(string txt);

int main(void) {
    string text = get_string("Text: ");
    check_grade(text);
}

void check_grade(string txt) {
    float letters = 0;
    float words = 1;
    float sentences = 0;

    int txtlen = strlen(txt);
    for (int i = 0; i < txtlen; i++) {
        if (isupper(txt[i]) || islower(txt[i])) {
            letters++;
        }
        else if (txt[i] == ' ') {
            words++;
        }
        else if (txt[i] == '.' || txt[i] == '!' || txt[i] == '?') {
            sentences++;
        }
    }

    float L = letters / (words / 100);
    float S = sentences / (words / 100);

    int index = round(0.0588 * L - 0.296 * S - 15.8);

    if (index < 1) {
        printf("Before Grade 1\n");
    }
    else if (index >= 1 && index <= 16) {
        printf("Grade %i\n", index);
    }
    else {
        printf("Grade 16+\n");
    }
}
